fs/vfs/fs_dup2.c

```diff
@@ -82,29 +81,15 @@ int file_dup3(FAR struct file *filep1, FAR struct file *filep2, int flags)
   inode = filep1->f_inode;
   inode_addref(inode);
 
-  /* If there is already an inode contained in the new file structure,
```

Comment:
how to handle this situation, realloced by others ?

Reply:
Can you please explain the use case how / where inode is realloced ? Is this solution "free without clear" still needed ?

Reply:
Suggest fully understand the code and then re-design the logic.

Here is the situation,
```
thread0                                               thread1
nx_dup2(fd1, fd2)
file_dup3(filep1, filep2)
close filep2
filep2->f_inode = NULL;
**means others can alloc this filep**
                                                        open, alloc a newfd
                                                        unfortunatly, alloc fd2
                                                        read & write from fd2.
opreated filep2, error
```
Hope to find a new method to resolve this.

Reply:
This race should not be possible after these patches, as filep and fd are separated. You can operate on filep even if the fd is freed.

Reply:
Current your patches has no this race, cause of lost close fd2 in nx_dup3_from_tcb()

from the spec:
   dup2()
       The dup2() system call performs the same task as dup(), but instead of using the lowest-numbered unused file descriptor, it uses the file descriptor number specified in newfd.  If the file descriptor newfd was previously open, it  is
       silently closed before being reused.